lib/diaspora_federation/validators/relayable_validator.rb

```diff
@@ -0,0 +1,11 @@
+module DiasporaFederation
+  module Validators
+    module RelayableValidator
+      def self.included(model)
+        model.class_eval do
+          rule :parent_guid, :guid
```

Comment:
I think you're probably deeper into the related code atm that I am. Did you compare behaviour between

* parent author is local
* parent author is remote
* author is local
* author is remote (received object as pod hosting the parent author)